modules/network/front-door-web-application-firewall-policy/.test/common/main.test.bicep

```diff
@@ -0,0 +1,128 @@
+param enableDefaultTelemetry bool = true
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '[[namePrefix]]'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2022-09-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-nestedDependencies'
+  params: {
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../main.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-test-${serviceShort}'
+  params: {
+    enableDefaultTelemetry: enableDefaultTelemetry
+    name: '${namePrefix}${serviceShort}001'
+    policySettings: {
+      mode: 'Prevention'
+      redirectUrl: 'http://www.bing.com'
+      customBlockResponseStatusCode: 200
+      customBlockResponseBody: 'PGh0bWw+CjxoZWFkZXI+PHRpdGxlPkhlbGxvPC90aXRsZT48L2hlYWRlcj4KPGJvZHk+CkhlbGxvIHdvcmxkCjwvYm9keT4KPC9odG1sPg=='
+    }
+    customRules: {
+      rules: [
+        {
+          name: 'CustomRule1'
+          priority: 2
+          enabledState: 'Enabled'
+          action: 'Block'
+          ruleType: 'MatchRule'
+          rateLimitDurationInMinutes: 1
+          rateLimitThreshold: 10
+          matchConditions: [
+            {
+              matchVariable: 'RemoteAddr'
+              selector: null
+              operator: 'GeoMatch'
+              negateCondition: false
+              transforms: []
+              matchValue: [
+                'CH'
+              ]
+            }
+            {
+              matchVariable: 'RequestHeader'
+              selector: 'UserAgent'
+              operator: 'Contains'
+              negateCondition: false
+              transforms: []
+              matchValue: [
+                'windows'
+              ]
+            }
+            {
+              matchVariable: 'QueryString'
+              operator: 'Contains'
+              negateCondition: false
+              transforms: [
+                'UrlDecode'
+                'Lowercase'
+              ]
+              matchValue: [
+                '<?php'
+                '?>'
+              ]
+            }
+          ]
+        }
+      ]
+    }
+    managedRules: {
+      managedRuleSets: [
+        {
+          ruleSetType: 'Microsoft_BotManagerRuleSet'
+          ruleSetVersion: '1.0'
+        }
+      ]
+    }
+    tags: {
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+    }
+    roleAssignments: [
+      {
+        roleDefinitionIdOrName: 'Reader'
+        principalIds: [
+          nestedDependencies.outputs.managedIdentityPrincipalId
+        ]
+        principalType: 'ServicePrincipal'
+      }
+    ]
+  }
```

Comment:
Can we include the lock and SKU properties as part of this test file?

Reply:
done